### PR TITLE
Creating a tools crate for both trusted and untrusted sides

### PR DESCRIPTION
**Pull request by elichai**
This moves `km_primitives` from `enigma-tools-t` to `enigma-tools-m` so that the untrusted side can use that too.
it does the same for `keeper_types` so that it won't be duplicated.

### CODE REVIEW

**Comment by elichai** [comment]
I reverted this commit because it made no sense to me and it broke stuff. https://github.com/enigmampc/enigma-core/pull/89/commits/1cd35e3a2175d9024416fcf25f9f8eec9077eb89
reversion commit: https://github.com/enigmampc/enigma-core/pull/99/commits/df9c4420161a7d8b7c3164cb493792bed59e80ed

**Comment by elichai** [comment]
triggering the ci

File: enigma-tools-m/src/primitives/km_primitives.rs [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+    pub fn get_pubkey(&self) -> PubKey {
+        let mut pubkey = [0u8; 64];
+        pubkey.copy_from_slice(&self.pubkey[..]);
+        pubkey
+    }
+
+    pub fn get_id(&self) -> MsgID { self.id }
+
+    pub fn is_request(&self) -> bool {
```
Comment (moriaab) [inline enigma-tools-m/src/primitives/km_primitives.rs:82]: I didn't see that this `is_<type>` functions are used anywhere. If they are indeed not used, it is better to remove them.

Comment (elichai) [reply enigma-tools-m/src/primitives/km_primitives.rs:82]: `is_response` is used https://github.com/enigmampc/enigma-core/blob/159a9b6c2f3e444b9380ad5da13ed4173d772a51/enigma-tools-m/src/primitives/km_primitives.rs#L39 but `is_request`/`is_encrypted_response` aren't used.

File: enigma-tools-m/src/primitives/serde_impls.rs [hunk 2]
```diff
@@ -0,0 +1,224 @@
+use crate::localstd::fmt::{self, Formatter};
+use crate::localstd::vec::Vec;
+use crate::primitives::km_primitives::*;
+use crate::serde::{de::{EnumAccess, Error, IgnoredAny, MapAccess, SeqAccess, Unexpected, VariantAccess, Visitor},
+                   ser::SerializeStruct,
+                   Deserialize, Deserializer, Serialize, Serializer};
+use enigma_types::{ContractAddress, StateKey};
+// The main reason why we need to implement Serialize/Deserialize ourselves is because the derive macro
```
Comment (moriaab) [inline enigma-tools-m/src/primitives/serde_impls.rs:8]: Great comment!

**Review by moriaab — APPROVED** [review]

**Review by fredfortier — APPROVED** [review]
Looks good to me.

**Pull request merged**